pkg/build/util/generate.go

```diff
@@ -30,3 +32,44 @@ func GenerateBuildFromBuild(build *api.Build) *api.Build {
 		},
 	}
 }
+
+// SubstituteImageReferences replaces references to an image with a new value
+func SubstituteImageReferences(build *api.Build, oldImage string, newImage string) {
+	switch {
```

Comment:
I can't do the && logic in the case statement if i do what you're proposing, so i'd have to do:
switch build.Parameters.Strategy.Type {
    case api.DockerBuildStrategyType:
         if otherchecks {
           ....
         }
    case api.STIBuildStrategyType:
        ...
    case api.CustomBuildStrategyType:
        ...
    }

which seems silly.  i think i'd prefer to leave it as is, or go back to an if/else if structure like i originally had it until @mfojtik weighed in. :)